reconcile/openshift_limitranges.py

```diff
@@ -0,0 +1,109 @@
+import logging
+import semver
+import sys
+
+import utils.gql as gql
+import reconcile.openshift_base as ob
+
+from utils.openshift_resource import OpenshiftResource as OR
+from utils.defer import defer
+
+from reconcile.queries import NAMESPACES_QUERY
+
+QONTRACT_INTEGRATION = 'openshift-limitranges'
+QONTRACT_INTEGRATION_VERSION = semver.format_version(0, 1, 0)
+
+SUPPORTED_LIMITRANGE_TYPES = (
+    'default',
+    'defaultRequest',
+    'max',
+    'maxLimitRequestRatio',
+    'min'
+)
+
+
+def construct_resources(namespaces):
+    for namespace in namespaces:
+        # Get the linked limitRanges schema settings
+        limitranges = namespace.get("limitRanges", {})
+
+        body = {
+            'apiVersion': 'v1',
+            'kind': 'LimitRange',
+            'metadata': {
+                'name': limitranges['name'],
+            },
+            'spec': {
+                'limits': []
+            }
+        }
+
+        # Build each limit item ignoring null ones
+        speclimit = {}
+        for l in limitranges['limits']:
+            for ltype in SUPPORTED_LIMITRANGE_TYPES:
+                if ltype in l and l[ltype] is not None:
+                    speclimit[ltype] = l[ltype]
+            body['spec']['limits'].append(speclimit)
+
+        resource = OR(body, QONTRACT_INTEGRATION, QONTRACT_INTEGRATION_VERSION)
+
+        # Create the resources and append them to the namespace
+        namespace["resources"] = [resource]
+
+    return namespaces
+
+
+def add_desired_state(namespaces, ri):
+    for namespace in namespaces:
+        for resource in namespace["resources"]:
```

Comment:
since we are modifying the existing namespaces, may this be uninitialized?
sorry if i'm repeating this too much and missing something obvious...

Reply:
You're right. I've added a key check